frame/elections-phragmen/src/lib.rs

```diff
@@ -890,222 +922,22 @@ impl<T: Config> Pallet<T> {
 		debug_assert!(_remainder.is_zero());
 	}
 
-	/// Run the phragmen election with all required side processes and state updates, if election
-	/// succeeds. Else, it will emit an `ElectionError` event.
+	/// Run the election with all required side processes and state updates, if election succeeds.
+	/// Else, it will emit an `ElectionError` event.
 	///
 	/// Calls the appropriate [`ChangeMembers`] function variant internally.
-	fn do_phragmen() -> Weight {
```

Comment:
To keep the diff easier to manage, I would suggest that you keep this function mostly intact, and only replace 

```
sp_npos_elections::seq_phragmen(num_to_elect, candidate_ids, voters_and_votes, None)
				.map(|ElectionResult::<T::AccountId, Perbill> { winners, assignments: _ }| {
``` 

with a call to `ElectionProvider`. Ofc the returning types are different and there needs to be a small conversion. I leave it up to you for now to figure that out.